### PR TITLE
:broom: chore: drop go 1.16 support & update dependencies

### PR DESCRIPTION
**Pull request by efectn**
## Description
1.16 is no longer supported by Fasthttp, let's drop it.

## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist:

- [ ] For new functionalities I follow the inspiration of the express js framework and built them similar in usage
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation - /docs/ directory for https://docs.gofiber.io/
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] If new dependencies exist, I have checked that they are really necessary and agreed with the maintainers/community (we want to have as few dependencies as possible)
- [ ] I tried to make my code as fast as possible with as few allocations as possible
- [ ] For new code I have written benchmarks so that they can be analyzed and improved

## Commit formatting:

Use emojis on commit messages so it provides an easy way of identifying the purpose or intention of a commit. Check out the emoji cheatsheet here: https://gitmoji.carloscuesta.me/

### CODE REVIEW

**Comment by gaby** [comment]
Go `1.17` is EOL since August 2022, would it make more sense to make the jump to `1.18` as minimum like the contrib/storage middlewares?

https://endoflife.date/go

**Review by gaby — APPROVED** [review]

File: prefork_test.go [hunk 1]
```diff
@@ -71,7 +71,6 @@ func Test_App_Prefork_Master_Process(t *testing.T) {
 }
 
 func Test_App_Prefork_Child_Process_Never_Show_Startup_Message(t *testing.T) {
-	t.Parallel()
```
Comment (leonklingele) [inline prefork_test.go:74]: This is most likely an unrelated change and should be reverted.

**Comment by gaby** [comment]
> Go `1.17` is EOL since August 2022, would it make more sense to make the jump to `1.18` as minimum like the contrib/storage middlewares?
> 
> https://endoflife.date/go

I also noticed that action/setup-go tags stable as 1.19 and oldstable as 1.18. We can probably add 1.19 to the test suite. 

Src: https://github.com/actions/setup-go#using-stableoldstable-aliases

**Comment by ReneWerner87** [comment]
> Go `1.17` is EOL since August 2022, would it make more sense to make the jump to `1.18` as minimum like the contrib/storage middlewares?
> 
> https://endoflife.date/go

we always support the latest version which supports fasthttp, in the new version of fiber we already make this jumpwe always support the latest version which supports fasthttp, in the new version of fiber we already make this jump

File: prefork_test.go (hunk 1 above)
Comment (efectn) [reply prefork_test.go:74]: Just try to fix data race on tests

**Review by ReneWerner87 — APPROVED** [review]
